src/library/searchquery.cpp

```diff
@@ -471,6 +471,263 @@ double DurationFilterNode::parse(const QString& arg, bool* ok) {
     return 60 * m + s;
 }
 
+// static
+constexpr double BpmFilterNode::kRelativeRangeDefault;
+
+// static
+double BpmFilterNode::s_relativeRange = kRelativeRangeDefault;
+
+// static
+void BpmFilterNode::setBpmRelativeRange(double range) {
+    // range < 0 would yield zero results because m_dRangeLow > m_dRangeHigh
+    VERIFY_OR_DEBUG_ASSERT(range >= 0) {
+        return;
+    }
+    s_relativeRange = range;
+}
+
+namespace {
+
+inline QString rangeSqlString(double lower, double upper) {
+    // 'BETWEEN' is inclusive, i.e. returns true if lower <= value <= upper
+    return QStringLiteral("bpm BETWEEN %1 AND %2")
+            .arg(QString::number(lower),
+                    QString::number(upper));
+}
+
+inline std::pair<double, double> rangeFromTrailingDecimal(double bpm) {
+    // Set up a range if we have decimals. This will include matches
+    // for which we show rounded values in the library. For example
+    // 124  finds 119.95 - 124.05
```

Comment:
Is this example still correct here?

Reply:
nope.
I'm about to clean up and add some examples in comments to make the modes and ranges easier to grasp.